### PR TITLE
[Improvement][Task] DataX Node doesn't Support Clickhouse as datasource (#5180)

### LINKED ISSUE #5180: [Improvement][Module Name] DataX Node doesn't Support Clickhouse as the datasource
**Describe the question**
When I create a DataX node, I can't use a Clickhouse Datasource as the source or the target in a DataX task. Currently, I can only select from Mysql, Postgresql，Oracle ans Sqlserver.

image

**Describe alternatives you've considered**
I hope I can't use Clickhouse as my datasource.

### LINKED ISSUE #5241: Bug: create dataX job json error
[https://github.com/apache/incubator-dolphinscheduler/blob/dev/dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/datax/DataxTask.java](url)
line:313-314
Reader and Writer in datax job json file is json string. DataX parse job json file failed.

### PR DESCRIPTION
**Pull request by ruanwenjun**
## Purpose of the pull request
This closes #5180, closes #5241

### CODE REVIEW

**Comment by CalvinKirs** [comment]
please add ut verify

**Comment by chengshiwen** [comment]
@ruanwenjun Could you Improve the `Coverage on New Code` to 33.3% at least? Thanks so much!

**Review by break60 — APPROVED** [review]
front end +1

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/datax/DataxTask.java [hunk 1]
```diff
@@ -462,7 +462,10 @@ public String loadJvmEnv(DataxParameters dataXParameters) {
 
         try {
             SQLStatementParser parser = DataxUtils.getSqlStatementParser(dbType, sql);
-            notNull(parser, String.format("database driver [%s] is not support", dbType.toString()));
+            if (parser == null) {
+                logger.warn("database driver [{}] is not support grammatical analysis sql", dbType);
+                return new String[0];
```
Comment (wen-hemin) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/datax/DataxTask.java:467]: Why not "return null;"

**Review by wen-hemin — CHANGES_REQUESTED** [review]
@ruanwenjun reviewed

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/datax/DataxTask.java (hunk 1 above)
Comment (ruanwenjun) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/datax/DataxTask.java:467]: I think there is no problem returning null, but the code smell check suggest we should use empty array to replace null.

**Review by wen-hemin — APPROVED** [review]
+1

**Pull request merged**